### ISSUE TITLE
Food Finder - Food Banks

### ISSUE THREAD

**Issue by karengreene** [midohiofoodbank/freshtrak-public#37]
As a customer, I want to find available food banks that serve my zip code so that I can get the food and supplies I need. **This application needs to be web and mobile friendly.**

--------

## Problem

Currently, in order to find an agency with available food resources, customers rely on word of mouth recommendations and a combination of 3rd-party listings. An existing issue with this system is that it is prone to errors, and slow to update - a reality that will only get worse due to the COVID-19 situation. Foodbanks and their agencies around the nation are also preparing for increased demand (more people experiencing layoffs, lost shifts, etc…), decreased supply of food resources (>50% of agencies are run by individuals 60+ years old and may close), as well as attempts to replace supply of lost food resources with the introduction of new food resource distribution models (e.g. delivery of groceries, mass distributions at fairgrounds, etc…). These changes in food resource availability will be difficult to navigate even for experienced foodbank staff, much less the average customer who is experiencing an initial introduction to the feeding network.

--------------------------

## Solution

The goal of this story is to return a list of food banks that serve the zip code searched.  

If a zip code is in a service territory we do not have agency data for:
- If the foodbank’s website is documented, redirect the customer to the foodbank’s website. E.g. A Cleveland zip code would direct the customer to [Partner Agencies ](https://www.greaterclevelandfoodbank.org/partner-agencies/list)
- If the foodbank’s website is not documented, redirect the customer to Feeding America’s “Find Your Local Foodbank” tool [Find Your Local Food Bank ](https://www.feedingamerica.org/find-your-local-foodbank)

---------------------------------------------------

## Screen Summary

The main screen lists food banks and pantries that serve that zip code. The scope of this story is the 'Food Bank' (first section) of this screen.  The main screen includes:
1. Food Banks - sorted alphabetically.
2. Resource Events - See issue #1 
3. Additional Resources - See issue #38 

#####################################################

**No pantries or Food Banks**
If there are no pantries or food banks that serve that zip code, display feeding america website.

------------------------------------------------------
## Screen Inputs

Zip Code - valid five digit zip code

---------------------------------------------------------
## Screen Details

The screen should list the following information for each food bank sorted alphabetically by name:
- Name of food bank
- Address 
- Website of food bank
- Phone Number

**Note that there are items on the wire frames that are out of scope for this story. See the Wireframes section for details.**

------------------------------------------------------------------------

## Wire frames

**Out of Scope**

**These wire frames show features that are NOT in scope for this story** and will be done at some later time. I don't have edit privileges for these wire frames, so I also listed these below.
 
 For the purpose of this story, ONLY focus on the 'Search for Resources' functionality. In this feature, the wire frames show the **below items which will NOT be included in the MVP**. These include:
- Address text box in the 'Find a location'
- Sort By
- Pre-Register
- Reserve Time
- Create Account
- Register Food Bank
- Sign In

###########################

**Wire Frame Links**

Wireframes in a mobile-experience view: https://www.figma.com/proto/R15ILgPqHRLhakIX9qIqQR/MOFC-MVP?node-id=237%3A828&scaling=scale-down

Wireframes in an Expanded Design View: https://www.figma.com/file/R15ILgPqHRLhakIX9qIqQR/MOFC-MVP?node-id=244%3A3009

----------------------------------------------------------------
## Acceptance Criteria

**#####  Testing Food Banks #####################################**

**AC** - Test food banks listed BEFORE pantries 
Given that food banks exist for zip code served
When results are displayed
Then food banks are displayed in 'Food Bank' section sorted alphabetically

**AC** - Test food banks results are ones that serve zip code
Given that food banks exist for zip code served
When food bank list is displayed
Results include all food banks that serve that zip code

**AC** - Test food banks details displayed
Given that food banks exist for zip code served
When food bank list is displayed
Then food bank name, address, phone, and website is displayed for each food bank

**AC** - Test food bank web site link
Given that resource search returns results 
And  website link is shown
When link is clicked
Then link opens food bank web site 

**AC** - If error retrieving data, user gets try again error
Given that the user has launched “Find Food Tool”
When there is an error in retrieving agency data
Then "Error try again" will be displayed

**AC** - invalid zip code entered
Given that the user has launched “Find Food Tool”
When a invalid zip code is entered
Then "please enter a valid zip code" will be displayed

**AC** - If there are no pantries or food banks, user gets feeding america website
Given that a user has launched “Find Food Tool”
When a user submits a valid zip code
And there are no agencies or food banks that serve that zip code
Then display a link to the Feeding America website (Feeding America’s “[Find Your Local Food Bank ](https://www.feedingamerica.org/find-your-local-foodbank)” tool Find Your Local Food Bank )

**Comment by PhilNorman2** [comment]
Ran some Foodbank test cases in https://beta.freshtrak.com/.  There were a couple of problems with displaying FB URLs.  I suspect the problem might be in the UI and I have contacted Sean.

enter zip code **45378**
result: Pass - Returns two foodbanks (Shared Harvest and The Foodbank, Inc. (Dayton) as expected.  The correct URL links are displayed.

enter zip code: **93036**
result: Fail - Expect The URL link https://www.feedingamerica.org/find-your-local-foodbank to be displayed, returned nothing

enter zip code: **15110**
result: Fail - Expect the URL https://www.pittsburghfoodbank.org/get-help/locator/ to be displayed, returned nothing

enter zip code: **14903**
result: Pass - The correct FB metadata and URL link are returned.
